src/react/components/common/tagsInput/tagsInput.tsx

```diff
@@ -146,11 +146,19 @@ export default class TagsInput<T extends ITagsInputProps> extends React.Componen
         return match;
     }
 
+    /**
+     * Get the selected tag name from event and remove index if present
+     * @param event onClick or keyPress event
+     */
     protected getTagText(event): string {
```

Comment:
Instead of parsing out the text we could also set a different attr on the element... either use the id field or set a `data-` attribute that we can lookup later.

Reply:
I'd also like to see a unit test validating this fix.

Reply:
Will do. The main struggle has been the inconsistency in the location of the text that I'm looking for. It's one place during tests and another when using the tool. I'll dive deeper on this and get it validated with some tests